e2e/nomostest/gitproviders/bitbucket.go

```diff
@@ -93,154 +97,155 @@ func (b *BitbucketClient) SyncURL(name string) string {
-	localName = strings.ReplaceAll(localName, "/", "")
-	localName = strings.ReplaceAll(localName, "-", "")
-	if len(localName) > localNameLength {
-		localName = localName[:localNameLength]
-	}
-
-	uuid := strings.ReplaceAll(u.String(), "-", "")
-
-	repoName := localName + uuid
-	if len(repoName) > repoNameMaxLength {
-		repoName = repoName[:repoNameMaxLength]
-	}
+func (b *BitbucketClient) CreateRepository(name string) (string, error) {
+	fullName := b.fullName(name)
+	repoURL := fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s/%s", b.workspace, fullName)
 
 	// Create a remote repository on demand with a random localName.
 	accessToken, err := b.refreshAccessToken()
 	if err != nil {
 		return "", err
 	}
 
-	out, err := exec.Command("curl", "-sX", "POST",
-		"-H", "Content-Type: application/json",
-		"-H", fmt.Sprintf("Authorization:Bearer %s", accessToken),
-		"-d", fmt.Sprintf("{\"scm\": \"git\",\"project\": {\"key\": \"%s\"},\"is_private\": \"true\"}", bitbucketProject),
-		fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s/%s", b.workspace, repoName)).CombinedOutput()
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
 
+	req, err := http.NewRequest(http.MethodGet, repoURL, nil)
 	if err != nil {
-		return "", fmt.Errorf("%s: %w", string(out), err)
-	}
-	if strings.Contains(string(out), "\"type\": \"error\"") {
-		return "", errors.New(string(out))
+		return "", fmt.Errorf("error creating request: %w", err)
 	}
-	return repoName, nil
-}
 
-// DeleteRepositories calls the DELETE API to delete all remote repositories on Bitbucket.
-// It deletes multiple repos in a single function in order to reuse the access_token.
-func (b *BitbucketClient) DeleteRepositories(names ...string) error {
-	accessToken, err := b.refreshAccessToken()
+	// Add custom headers to the request
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	req.Header.Set("Content-Type", "application/json")
+
+	// Execute the request
+	resp, err := client.Do(req)
 	if err != nil {
-		return err
+		return "", fmt.Errorf("error sending request: %w", err)
 	}
 
-	return b.deleteRepos(accessToken, names...)
-}
-
-// TODO: This is a temporary workaround for the known issue
-// go/acm-e2e-testing#bitbucket-repository-deletion-failure
-// Until this is fixed, we are skipping removal of repos stuck in bad state
-func (b *BitbucketClient) deleteRepos(accessToken string, names ...string) error {
-	var errs error
-	for _, name := range names {
-		_, err := retry.Retry(30*time.Second, func() error {
-			return b.deleteSingleRepo(accessToken, name)
-		})
-		if err != nil {
-			// Skip repositories that are currently unavailable
-			if strings.Contains(err.Error(), "Repository currently not available") {
-				fmt.Printf("[BITBUCKET] Skipping repository %s: %v\n", name, err)
-				continue
-			}
-			errs = multierr.Append(errs, err)
+	if resp.StatusCode == http.StatusOK { //Already exists
```

Comment:
In the case that the repo already exists, is there logic in place to ensure a clean initial state of the repository? For example a hard reset / force push to a known initial state